### ISSUE TITLE
Dialog: dialogs will disappear upon any click if opened from an OverlayPanel

### ISSUE THREAD

**Issue by matthew-dean** [primefaces/primevue#5256]
### Describe the bug

A dialog opened from an OverlayPanel is instantly dismissed when anywhere on the screen is clicked.

### Reproducer

https://stackblitz.com/edit/primevue-create-vue-typescript-issue-template-vfkvmz?file=src%2FApp.vue

### PrimeVue version

3.47.2

### Vue version

3.x

### Language

TypeScript

### Build / Runtime

Vue CLI App

### Browser(s)

any

### Steps to reproduce the behavior

1. Have a button that opens an OverlayPanel
2. Within that button, have another button and Dialog. The 2nd button opens the dialog
3. Open the dialog
4. Click anywhere
5. Watch the dialog disappear
6. **Second bug** If the OverlayPanel is re-opened, the dialog will start in an open state.

### Expected behavior

There is a developer expectation that dialogs behave somewhat like independent windows, and whatever opens them is irrelevant to their state, unless specified. You can test this / verify this by using a native browser prompt or any dialog from any app. For example, say you have an app that has a drop-down menu (which is my use case). If you select an option that opens a dialog, that app menu will likely disappear, yet the dialog will, of course, persist.

EDIT: Another way to think about this is that even if we think of the Dialog as a "child" of the OverlayPanel (and not an independent window), then it also logically follows that any click on the Dialog / Dialog bg is still a click "within" the OverlayPanel and therefore should not close the OverlayPanel. (Yet I still might argue that even if the OverlayPanel is programmatically closed, it might still be a violation of the Law of Least Surprise to immediately close / detach a Dialog opened from the OverlayPanel.)

**Comment by mertsincan** [comment]
Thanks a lot for your report! I set a milestone for it. We'll check it before the milestone is released.

**Comment by Tefoh** [comment]
I could have had a workaround for this issue, and I used the `dismissable` prop on the overlay component. I checked the code and found that this event had been set globally, causing this issue.
https://github.com/primefaces/primevue/blob/v3-prod/components/lib/overlaypanel/OverlayPanel.vue#L222

I didn't need overlay to be dismissable so this worked for me:
```vue
  <OverlayPanel ref="panel" :dismissable="false">
    <Button label="Open Dialog" @click="dialogVisible = true"></Button>
    <Dialog v-model:visible="dialogVisible" modal> I am a dialog </Dialog>
  </OverlayPanel>
```

 If I need the dismissable, I can set the event conditionally based on whether the overlay is shown or not.
